superset-frontend/src/dashboard/actions/nativeFilters.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ExtraFormData, SupersetClient } from '@superset-ui/core';
+import { Dispatch } from 'redux';
+import {
+  Filter,
+  FilterConfiguration,
+  SelectedValues,
+} from '../components/nativeFilters/types';
+import { dashboardInfoChanged } from './dashboardInfo';
+
+export const SET_FILTER_CONFIG_BEGIN = 'SET_FILTER_CONFIG_BEGIN';
+export interface SetFilterConfigBegin {
+  type: typeof SET_FILTER_CONFIG_BEGIN;
+  filterConfig: FilterConfiguration;
+}
+export const SET_FILTER_CONFIG_COMPLETE = 'SET_FILTER_CONFIG_COMPLETE';
+export interface SetFilterConfigComplete {
+  type: typeof SET_FILTER_CONFIG_COMPLETE;
+  filterConfig: FilterConfiguration;
+}
+export const SET_FILTER_CONFIG_FAIL = 'SET_FILTER_CONFIG_FAIL';
+export interface SetFilterConfigFail {
+  type: typeof SET_FILTER_CONFIG_FAIL;
+  filterConfig: FilterConfiguration;
+}
+
+export const SET_FILTER_STATE = 'SET_FILTER_STATE';
+export interface SetFilterState {
+  type: typeof SET_FILTER_STATE;
+  selectedValues: SelectedValues;
+  filter: Filter;
+  filters: FilterConfiguration;
+}
+
+export const setFilterConfiguration = (
+  filterConfig: FilterConfiguration,
+) => async (dispatch: Dispatch, getState: () => any) => {
+  dispatch({
+    type: SET_FILTER_CONFIG_BEGIN,
+    filterConfig,
+  });
+  const { id, metadata } = getState().dashboardInfo;
+  try {
+    const response = await SupersetClient.put({
+      endpoint: `/api/v1/dashboard/${id}`,
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        json_metadata: JSON.stringify({
+          ...metadata,
+          filter_configuration: filterConfig,
+        }),
+      }),
+    });
```

Comment:
Could use `makeApi` so you don't have to manually pass the headers and parse the body.

```ts
import { makeApi, JsonValue } from '@superset-ui/core';

interface DashboardInfo {
  id: number;
  json_metadata: string;
  // ... other fields of the Dashboard API resource
}

const updateDashboard = makeApi<Partial<DashboardInfo>, { result: DashboardInfo }>({
  method: 'PUT',
  endpoint: `/api/v1/dashboard/${id}`
});

const { json_metadata } = await updatedFilterConfig({
  json_metadata: JSON.stringify({
    ...metadata,
    filter_configuration: filterConfig,
  }),
});
```

Reply:
Interesting, I'll look into it.

Reply:
Just a note, while making this change I noticed that `makeApi` doesn't appear to support parameterized urls so you have to call `makeApi` at the call site where you know what the id is. Might take a stab at changing that later.